client-java/controller/src/main/java/org/evomaster/client/java/controller/problem/rpc/RPCEndpointsBuilder.java

```diff
@@ -1313,80 +1313,134 @@ private static int getDepthLevel(Class clazz, List<String> flattendepth, int lev
-                                        if (actionDto.mockDatabaseDtos.get(i) == null)
-                                            SimpleLogger.recordErrorMessage(
-                                                    String.format("Seeded Test Error: specify null mockDatabaseDto at index %d for action %s:%s", i, actionDto.interfaceName, actionDto.functionName)
-                                            );
-                                    }
-                                    rpcActionDto.mockDatabaseDtos = actionDto.mockDatabaseDtos.stream().filter(Objects::nonNull).collect(Collectors.toList());
-                                }
-                                handleExternalResponses(schema, actionDto, rpcType);
-                                test.add(rpcActionDto);
-                            }else {
-                                SimpleLogger.recordErrorMessage("Seeded Test Error: cannot find the action "+actionDto.functionName);
-                            }
-                        } else {
-                            SimpleLogger.recordErrorMessage("Seeded Test Error: cannot find the interface "+ actionDto.interfaceName);
-                        }
-                    }
                     String testKey = String.format("%s_INDEX_%d", (dto.testName != null)?dto.testName:"untitled", seedRPCTests.indexOf(dto));
-                    if (!test.isEmpty())
-                        results.put( testKey, test);
-                    else
+
+                    RPCTestDto testDto = handleSeededRPCFunctions(rpcInterfaceSchema, dto, rpcType, testKey);
+
+
+                    if (testDto != null && ((testDto.rpcFuctions != null && !testDto.rpcFuctions.isEmpty()) || (testDto.scheduleTaskInvocationDtos != null && !testDto.scheduleTaskInvocationDtos.isEmpty()))
+                    ) {
+                        results.put(testKey, testDto);
+                    } else
                         SimpleLogger.recordErrorMessage("Seeded Test Error: fail to load the seeded test "+ testKey);
                 }catch (RuntimeException e){
                     SimpleLogger.recordErrorMessage("Fail to handle specified seeded test: "+ ((dto.testName != null)? dto.testName:"index_"+seedRPCTests.indexOf(dto)));
                     StringBuilder msg = extractExceptionMsg(e);
                     SimpleLogger.recordErrorMessage(msg.toString());
                 }
+
             } else {
                 SimpleLogger.warn("Seeded Test: empty RPC function calls for the test "+ ((dto.testName != null)? dto.testName:"index_"+seedRPCTests.indexOf(dto)));
             }
         }
         return results;
     }
 
+
+    private static RPCTestDto handleSeededRPCFunctions(Map<String, InterfaceSchema> rpcInterfaceSchema,SeededRPCTestDto dto, RPCType rpcType, String testName) throws RuntimeException{
+        // handle RPC actions
+        List<RPCActionDto> test = null;
+        if (dto.rpcFunctions != null && !dto.rpcFunctions.isEmpty()){
```

Comment:
there is a lot of nesting in this function that makes it harder to read. best to invert the predicates, eg using things like `if (dto.rpcFunctions == null || dto.rpcFunctions.isEmpty()) return`  and `if(schema == null) continue`, which avoid too much nesting